.github/workflows/fastapi.yml

```
@@ -0,0 +1,42 @@
+name: fastapi-testsuite
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+jobs:
+  fastapi-test:
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        working-directory: fastapi
+    strategy:
+      matrix:
+        python-version: [3.6, 3.7, 3.8, 3.9]
+      fail-fast: false
+    steps:
+      - name: Set up Python
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python-version }}
+      - uses: actions/checkout@v2
+        with:
+          path: ddtrace
+      - uses: actions/checkout@v2
+        with:
+          repository: tiangolo/fastapi
+          ref: 0.63.0
+          path: fastapi
+      - name: Install Flit
+        run: pip install flit
+      - name: Install Dependencies
+        run: flit install --symlink
+      - name: Downgrade SQLalchemy
+        run: pip install --force-reinstall SQLAlchemy==1.3.24
+      - name: Install Typing
+        run: pip install typing
+      - name: Inject ddtrace
+        run: pip install ddtrace
```

```suggestion
        run: pip install ../ddtrace
```
We typically want to run the tests on the modified code being submitted in the PR instead of pulling from pypi 😄

---

Just updated to this code, now Python version 3.7 and 3.8 are breaking when installing dd-trace

---

@Yun-Kim

---

Seems like this is a `typing`-related issue that stems from `typing` being in the standard python library starting from 3.5, can see more here: https://github.com/python/typing/issues/573
What was the reason we added the install typing step?

---

@Yun-Kim Originally installed bc tests were failing locally without it. Will remove to see if this causes tests to succeed.

---

@Yun-Kim Tests are now succeeding for fastapi, but failing other tests (flask, circleci)

---

those two CircleCI ones are a known issue we are working through right now, see #2371

for the flask framework test suite, you can ignore for now, it has been a little flaky. something we need to resolve (that specific failure will not block this PR... CircleCI ones will though)

---

@brettlangdon thanks for the clarification! I will work on other tests while I wait for the Circleci issues to resolve.

---

should be resolved now if you update from `master`